app/generic/definitions.py

```diff
@@ -353,12 +353,23 @@ async def create_schema(
         schema_name=schema.name,
         schema_version=schema.version,
     )
-    result = await aries_controller.schema.publish_schema(
-        body=schema_send_request, create_transaction_for_endorser=False
-    )
-
-    if not isinstance(result, SchemaSendResult) or not result.schema_:
-        raise CloudApiException("Error creating schema", 500)
+    try:
+        result = await aries_controller.schema.publish_schema(
+            body=schema_send_request, create_transaction_for_endorser=False
+        )
+    except ClientResponseError as e:
+        if e.status == 400 and "already exist" in e.message:
+            schema_ids = await aries_controller.schema.get_created_schemas(
+                schema_name=schema.name
+            )
+            schema = await aries_controller.schema.get_schema(
+                schema_id=schema_ids.schema_ids[0]
+            )
```

Comment:
There could be multiple schema's with the same name. If the schema already exists we can construct the schema id based on the did, schema_name and schema_version. Otherwise we risk returning a different schema that is unrelated to the schema we were trying to create.

One thing also to note is that the attributes we passed to create the schema could different from the attributes that are present in the existing schema. We should probably do a check for this, or just throw an error like before that the schema already exists